src/agent.rs

```diff
@@ -594,6 +596,53 @@ impl AgentBuilder {
         self.cookie_store = Some(cookie_store);
         self
     }
+
+    /// Add middleware handler to this agent.
+    ///
+    /// All requests made by the agent will use this middleware. Middleware is invoked
+    /// in the order they are added to the builder.
+    pub fn middleware(mut self, m: impl Middleware) -> Self {
+        self.middleware.push(Box::new(m));
+        self
+    }
+}
+
+#[cfg(feature = "tls")]
+#[derive(Clone)]
+pub(crate) struct TLSClientConfig(pub(crate) Arc<rustls::ClientConfig>);
+
+#[cfg(feature = "tls")]
+impl fmt::Debug for TLSClientConfig {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("TLSClientConfig").finish()
+    }
+}
```

Comment:
These seem like an unrelated change?

Reply:
The debug stuff?

It's because deriving debug would require `Middleware` to also be debug, which felt unnecessary. 

The tls debug changes was because my preferred form is `impl fmt::Debug` and `use std::fmt` so I changed that one too so all are consistent in this file.